server/provider/errors.go

```diff
@@ -25,3 +26,23 @@ func newErrCannotGetAccount(address string, innerError error) error {
 func newErrCannotGetTransaction(hash string, innerError error) error {
 	return fmt.Errorf("%w: %v, address = %s", errCannotGetTransaction, innerError, hash)
 }
+
+// In elrond-proxy-go, the function CallGetRestEndPoint() returns an error message as the JSON content of the erroneous HTTP response.
+// Here, we attept to decode that JSON and create an error with a "flat" error message.
+func convertStructuredApiErrToFlatErr(apiErr error) error {
+	structuredApiErr := &structuredApiError{}
+	err := json.Unmarshal([]byte(apiErr.Error()), structuredApiErr)
+	if err != nil {
+		// Not parsable, fallback to original
+		return apiErr
+	}
+
+	flatErrString := fmt.Sprintf("%s; %s", structuredApiErr.Error, structuredApiErr.Code)
```

Comment:
maybe `%s: %s` . in practice:
`key not found; internal_error` vs `key not found: internal_error`
or it might be `internal_error: key not found`

Reply:
Fixed.